src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,214 @@
+        >
+          <TargetPathSelect
+            name={`${namePrefix}.targetPath`}
+            index={index}
+            choices={choices}
+            mappedVariable={mappedVariable}
+          />
+        </Field>
+      </FormRow>
+      <div style={{marginTop: '1em'}}>
+        <a href="#" onClick={e => e.preventDefault() || toggleJsonSchemaVisible()}>
+          <FormattedMessage
+            description="Objects API variable configuration editor JSON Schema visibility toggle"
+            defaultMessage="Toggle JSON Schema"
+          />
+        </a>
+        {jsonSchemaVisible && (
+          <pre>
+            {loading || !mappedVariable.targetPath
+              ? 'N/A'
+              : JSON.stringify(getTargetPath(mappedVariable.targetPath).jsonSchema, null, 2)}
+          </pre>
+        )}
+      </div>
+    </Fieldset>
+  );
+};
+
+ObjectsApiVariableConfigurationEditor.propTypes = {
+  variable: PropTypes.shape({
+    key: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+const TargetPathSelect = ({name, index, choices, mappedVariable}) => {
+  // To avoid having an incomplete variable mapping added in the `variablesMapping` array,
+  // It is added only when an actual target path is selected. This way, having the empty
+  // option selected means the variable is unmapped (hence the `arrayHelpers.remove` call below).
+
+  const {
+    values: {variablesMapping},
+    getFieldProps,
+    setFieldValue,
+  } = useFormikContext();
+  const props = getFieldProps(name);
+  const isNew = variablesMapping.length === index;
+
+  return (
+    <FieldArray
+      name="variablesMapping"
+      render={arrayHelpers => (
+        <Select
+          id="targetPath"
+          name={name}
+          allowBlank
+          choices={choices}
+          {...props}
+          value={JSON.stringify(props.value)}
+          onChange={event => {
+            if (event.target.value === '') {
+              arrayHelpers.remove(index);
+            } else {
+              if (isNew) {
+                arrayHelpers.push({...mappedVariable, targetPath: JSON.parse(event.target.value)});
+              } else {
+                setFieldValue(name, JSON.parse(event.target.value));
+              }
+            }
+          }}
+        />
+      )}
+    />
+  );
+};
+
+TargetPathSelect.propTypes = {
+  name: PropTypes.string.isRequired,
+  index: PropTypes.number.isRequired,
+  choices: PropTypes.array.isRequired,
+};
+
+const TargetPathDisplay = ({target}) => {
+  const path = target.targetPath.join(' > ');
+  return (
+    <FormattedMessage
+      description="Representation of a JSON Schema target path"
+      defaultMessage="{required, select, true {{path} (required)} other {{path}}}"
+      values={{path, required: target.isRequired}}
+    />
+  );
+};
+
+TargetPathDisplay.propTypes = {
+  target: PropTypes.shape({
+    targetPath: PropTypes.arrayOf(PropTypes.string).isRequired,
+  }).isRequired,
+  isRequired: PropTypes.bool.isRequired,
+};
```

Comment:
```suggestion
TargetPathDisplay.propTypes = {
  target: PropTypes.shape({
    targetPath: PropTypes.arrayOf(PropTypes.string).isRequired,
    isRequired: PropTypes.bool.isRequired,
  }).isRequired,
};
```
Maybe it was meant to be inside?